base/set.jl

```diff
@@ -474,3 +474,220 @@ end
 
 convert(::Type{Set{T}}, s::Set{T}) where {T} = s
 convert(::Type{Set{T}}, x::Set) where {T} = Set{T}(x)
+
+
+
+## replace/replace! ##
+
+# NOTE: to implement replace! and replace for a new type T, it's enough to define:
+# function _replace!(pred::Callable, A::T, count::Int)
```

Comment:
Again, I don't really like recommending people to overload an internal function. Is there a way to make this `replace!(pred::Callable, A::T; count::Int)`? BTW, keyword arguments are supposed to be faster now.

Reply:
Ah right, I didn't re-evaluate this choice in light of the supposedly faster keyword args. I would be willing to benchmark that, but as I'm very short on time now, I will accept your offer to address that later by giving it a lower priority.

Reply:
Why not go with the simplest solution of defining only the keyword argument version for now, and see later whether performance can be improved?